tests/test_cpp_interface.cpp

```diff
@@ -0,0 +1,68 @@
+#include <CppUTest/TestHarness.h>
+#include <CppUTestExt/MockSupport.h>
+#include <memory>
+#include "../messagebus.h"
+
+
+TEST_GROUP(MessagebusCppInterface)
+{
+    messagebus_topic_t raw_topic;
+    int topic_content;
+    messagebus_t bus;
+    std::unique_ptr<messagebus::TopicWrapper<int> > topic;
+    void setup(void)
+    {
+        messagebus_topic_init(&raw_topic, nullptr, nullptr, &topic_content, sizeof(topic_content));
+        messagebus_init(&bus, nullptr, nullptr);
+        messagebus_advertise_topic(&bus, &raw_topic, "/foo");
+        topic = std::make_unique<messagebus::TopicWrapper<int> >(&raw_topic);
+    }
+};
+
+TEST(MessagebusCppInterface, CanPublish)
+{
+    topic->publish(42);
+
+    int read_msg;
+    auto res = messagebus_topic_read(&raw_topic, &read_msg, sizeof(read_msg));
```

Comment:
I don't really like `auto` in UTs, I would use `bool res`, so we know what the function under test returns without referring to the header file (saves me two keystrokes #OptimizingMyLife).
[nitpicking] Missing newline

Reply:
Haha I have clang's type engine mapped in vim so only one keystroke gives me the type of the `auto` ;)